src/account/PluginManager2.sol

```diff
@@ -43,7 +42,9 @@ abstract contract PluginManager2 {
             for (uint256 i = 0; i < preValidationFunctions.length; ++i) {
                 FunctionReference preValidationFunction = preValidationFunctions[i];
 
-                _storage.validationData[validationFunction].preValidationHooks.push(preValidationFunction);
+                _storage.validationData[validationConfig.functionReference()].preValidationHooks.push(
```

Comment:
Lots of references to `_storage.validationData[validationConfig.functionReference()]`. Maybe assign a storage var to it.

Reply:
Good idea, updated